scripts/o3de/o3de/repo_properties.py

```diff
@@ -17,13 +17,142 @@
 import logging
 import hashlib
 import shutil
+import requests
+import sys
+import msvcrt
+import time
+import re
 from packaging.version import Version, InvalidVersion
 from packaging.specifiers import SpecifierSet
 from o3de import manifest, utils, validation
 
 logger = logging.getLogger('o3de.repo_properties')
 logging.basicConfig(format=utils.LOG_FORMAT)
 
+def hide_credential(prompt):
+    """
+    Hide credentials displays a prompt to the user and waits for credential input. The entered characters
+    are masked with asterisks (*) to provide privacy. The function captures individual keystrokes, including backspace,
+    and supports a timeout feature to limit the duration for user input.
+
+    Args:
+        prompt (str): The prompt to display to the user.
+
+    Returns:
+        str: The user-entered credential/password as a string.
+
+    Timeout:
+        If no input is received within 60 seconds, the function will display an
+        timeout message, return an empty string, and terminate.
+    """
+    sys.stdout.write(prompt)
+    sys.stdout.flush()
+
+    secret_input = []
+    start_time = time.time()
+
+    while True:
+        if msvcrt.kbhit():
+            char = msvcrt.getch()
+            if char == b'\r':
+                break
+            elif char == b'\x08':
+                if secret_input:
+                    secret_input.pop()
+                    sys.stdout.write('\b \b')
+            else:
+                sys.stdout.write('*')
+                secret_input.append(char)
+            sys.stdout.flush()
+            start_time = time.time()
+
+        if time.time() - start_time > 60:
+            sys.stdout.write('\n')
+            print("Input timeout occurred")
+            return ""
+
+    sys.stdout.write('\n')
+    return b''.join(secret_input).decode('utf-8')
+
+def upload_release_to_github(repo_uri:str,
+                             zip_path: pathlib.Path,
+                             archive_filename: str,
+                             upload_github_release_tag: str):
```

Comment:
lets move this github specific code to `github_utils.py` and make a `GitProvierInterface` function called `upload_release` that the `GitHubProvider` class implements with this specific code